### PR TITLE
Reduce and improve security logging

### PR DESCRIPTION
**Pull request by nielsm5**
Results in the following log line:
`
2023-09-29 12:00:21,728 [http-nio-80-exec-2] created bus request from URI [PUT:http://localhost/iaf-test/iaf/api/server/logging] issued by remoteHost [0:0:0:0:0:0:0:1] remoteAddress [0:0:0:0:0:0:0:1] remoteUser [Admin] with headers [enableDebugger=true, logLevel=WARN, maxMessageLength=-1, logIntermediaryResults=true] payload [NONE]
`

### CODE REVIEW

File: console/backend/src/main/java/nl/nn/adapterframework/management/web/RequestMessageBuilder.java [hunk 1]
```diff
@@ -85,7 +86,17 @@
...
-		SEC_LOG.always().log(createLogMessage());
+		if(SEC_LOG.isInfoEnabled()) {
+			String method = base.getServletRequest().getMethod();
+			if((method.equalsIgnoreCase("GET") || method.equalsIgnoreCase("OPTIONS"))) {
+				SEC_LOG.debug("created bus request from URI [{}:{}] issued by{}", method, base.getUriInfo().getRequestUri(), HttpUtils.getCommandIssuedBy(base.getServletRequest()));
+			} else {
+				String headers = customHeaders.entrySet().stream()
+						.map(e -> e.getKey() + "=" +e.getValue())
+						.collect(Collectors.joining (", "));
+				SEC_LOG.info("created bus request from URI [{}:{}] issued by{} with headers [{}] payload [{}]", method, base.getUriInfo().getRequestUri(), HttpUtils.getCommandIssuedBy(base.getServletRequest()), headers, payload);
```
[the remark this answers is not in the document]
Comment (tnleeuw) [reply console/backend/src/main/java/nl/nn/adapterframework/management/web/RequestMessageBuilder.java:97]: Sanitisation van de headers en payload nodig door alle \n en \r characters te vervangen door iets wat geen control-character is.

**Review by tnleeuw — CHANGES_REQUESTED** [review]

**Review by tnleeuw — APPROVED** [review]
I think the Sonar alerts can now be dismissed as false positives.

File: console/backend/src/main/java/nl/nn/adapterframework/management/web/RequestMessageBuilder.java [hunk 2]
```diff
@@ -85,7 +100,18 @@ public static RequestMessageBuilder create(FrankApiBase base, BusTopic topic, Bu
 	}
 
 	public Message<?> build() {
-		SEC_LOG.always().log(createLogMessage());
+		if(SEC_LOG.isInfoEnabled()) {
+			String method = base.getServletRequest().getMethod();
+			String issuedBy = sanitizeForLog(HttpUtils.getCommandIssuedBy(base.getServletRequest()));
+			if((method.equalsIgnoreCase("GET") || method.equalsIgnoreCase("OPTIONS"))) {
+				SEC_LOG.debug("created bus request from URI [{}:{}] issued by{}", method, base.getUriInfo().getRequestUri(), issuedBy);
```
Comment (jkosternl) [inline console/backend/src/main/java/nl/nn/adapterframework/management/web/RequestMessageBuilder.java:107]: A few points on the above code:
1. Line 103: `.isInfoEnabled` -> and you log on debug level at line 107 but on 112 on info level. Should it be `.isDebugEnabled` or logging should be on info-level at line 107?
2. The log text on line 107 and 112 are missing a white space at `by{}`

**Review by jkosternl — APPROVED** [review]

File: console/backend/src/main/java/nl/nn/adapterframework/management/web/RequestMessageBuilder.java (hunk 2 above)
Comment (nielsm5) [reply console/backend/src/main/java/nl/nn/adapterframework/management/web/RequestMessageBuilder.java:107]: 1. If loglevel == DEBUG, it also logs at INFO level. This check covers both levels.
2. the issuedBy string starts with a space.

Comment (jkosternl) [reply console/backend/src/main/java/nl/nn/adapterframework/management/web/RequestMessageBuilder.java:107]: Ok

**Pull request merged**